bin/codemods/src/single-tree-rendering.js

```diff
@@ -15,8 +31,25 @@ export default function transformer( file, api ) {
 		return str.replace( /(import.*\n)\n+(import)/g, '$1$2' );
 	}
 
+	/**
+	 * Ensure `context` is among parameters
+	 */
+	function ensureContextMiddleware( path ) {
+		// `context` param is already in
+		if ( path.value.params.indexOf( 'context' ) !== -1 ) {
+			return path.value;
+		}
+		let ret = path.value;
+		ret.params = [ j.identifier( 'context' ), ...ret.params ];
+
+		return ret;
+	}
+
+	/**
+	 * Adds `next` to params and `next()` to body
+	 */
 	function addNextMiddleware( path ) {
-		if ( path.value.params.length !== 1 ) {
+		if ( path.value.params.length > 1 ) {
```

Comment:
see https://github.com/Automattic/wp-calypso/pull/19567/files#r149681361

Reply:
Yep so `ensureContextMiddleware()` bit is run first and therefore when `addNextMiddleware()` runs, there's exactly one param (`context`).